x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/action/RestEsqlQueryActionTests.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.action;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.FakeRestRequest;
+import org.elasticsearch.xpack.esql.version.EsqlVersion;
+import org.hamcrest.Matcher;
+
+import java.util.List;
+
+import static org.elasticsearch.xpack.esql.action.RestEsqlQueryAction.CLIENT_META;
+import static org.elasticsearch.xpack.esql.action.RestEsqlQueryAction.PRODUCT_ORIGIN;
+import static org.elasticsearch.xpack.esql.action.RestEsqlQueryAction.defaultVersionForOldClients;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class RestEsqlQueryActionTests extends ESTestCase {
+    public void testNoVersionForNoClient() {
+        assertEsqlVersion(null, null, nullValue(String.class));
+    }
+
+    public void testNoVersionForAlreadySet() {
+        EsqlQueryRequest esqlRequest = new EsqlQueryRequest();
```

Comment:
nit/bikeshed: since we do not _really_ test the handling of RestEsqlQueryAction and the tests are based on EsqlQueryRequests anyway - we only really test , we could a) move the tests to EsqlQueryRequestsTets and b) make defaultVersionForOldClients a method of EsqlQueryRequest.

Reply:
I thought about this one and didn't want to. I like the defaulting method in the REST handler because it's an "incoming layer" hack sorta thing. And that implies sticking it in these tests. It'd be reasonably to put it in the request, but I'm the one typing so I'm going to keep it here.